### PR TITLE
Update README: Adding more details on how to instantiate openwhisk

### PR DESCRIPTION
**Pull request by pritidesai**
Adding more details on how to instantiate openwhisk.

Addressing some part of issue #21.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,10 +11,28 @@ $ npm install openwhisk
...
+	var base64Auth = new Buffer(auth).toString('base64');
+	var apiKey = "Basic " + base64Auth;
+
+	var options = {apihost: apiHost, api_key: apiKey, namespace: namespace};
+	
+	// instantiate the openwhisk instance before you can use it
+	var wsk = openwhisk(options);
+	...
+}
 ```
```
Comment (jthomas) [inline README.md]: In this OpenWhisk environment, the client library will automatically pull the credential values from environment values. The developer does not need to pass in any parameters. I should add an example to make this clear.

File: README.md [hunk 2]
```diff
@@ -11,10 +11,28 @@ $ npm install openwhisk
...
+	var openwhisk = require('openwhisk');
+
+	// read apihost, auth, and namespace from params
+	var apiHost = params.apiHost; # e.g. openwhisk.ng.bluemix.net
+	var namespace = params.namespace;
+	var auth = params.auth; 
+
+	// generate api_key from auth
+	var base64Auth = new Buffer(auth).toString('base64');
+	var apiKey = "Basic " + base64Auth;
```
Comment (jthomas) [inline README.md]: Did this example work for you? The user does not have to pass in the base64 string, the library handles that. It should be the raw auth string (user:pass).

File: README.md (hunk 1 above)
Comment (pritidesai) [inline README.md]: Yup, its my bad, didn't realize but tried without wrapping openwhisk() call inside of main(), which failed with "error": "Initialization has failed due to: Error: Invalid constructor options. Missing api_key parameter.
But, I guess the assumption here is node js action has to have main() function in .js file.

File: README.md (hunk 2 above)
Comment (pritidesai) [inline README.md]: Yup and found it weird (running on bluemix).

Without "return" on wsk.rules.list():
```
function main(params) {
    var openwhisk = require("openwhisk");

    var api_key = "user:pass";
    var base64_api_key = new Buffer(api_key).toString('base64');
    var encoded_api_key = "Basic " + base64_api_key;

    console.log(encoded_api_key)

    options = {
        api: "https://openwhisk.ng.bluemix.net/api/v1/",
        api_key: encoded_api_key,
        namespace: "my_namespace"
    }

    var wsk = openwhisk(options);

    wsk.rules.list()
    .then(function (packages) {
        console.log("PACKAGES");
        console.log(packages);
        return { packages: packages };
    })
    . catch(function (err) {
        console.log(err);
        return err;
    });
}
```
Result is:

 ```
   "response": {
        "status": "success",
        "statusCode": 0,
        "success": true,
        "result": {}
    },
```
With "return" wsk.rules.list():

```
function main(params) {
    var openwhisk = require("openwhisk");

    var api_key = "user:pass";
    var base64_api_key = new Buffer(api_key).toString('base64');
    var encoded_api_key = "Basic " + base64_api_key;

    console.log(encoded_api_key)

    options = {
        api: "https://openwhisk.ng.bluemix.net/api/v1/",
        api_key: encoded_api_key,
        namespace: "my_namespace"
    }

    var wsk = openwhisk(options);

    return wsk.rules.list()
    .then(function (packages) {
        console.log("PACKAGES");
        console.log(packages);
        return { packages: packages };
    })
    . catch(function (err) {
        console.log(err);
        return err;
    });
}
```
Result is:

```
  "response": {
        "status": "application error",
        "statusCode": 0,
        "success": false,
        "result": {
            "error": {
                "code": 3.11923e+06,
                "error": "The supplied authentication is invalid"
            }
        }
    },
```

File: README.md (hunk 1 above)
Comment (jthomas) [inline README.md]: Aha - this is due to the following bug 
https://github.com/openwhisk/openwhisk/issues/1751
I'm going to update the README to make this behaviour more explicit.

File: README.md (hunk 2 above)
Comment (jthomas) [inline README.md]: Okay I understand what's happening. 

Without the `return` statement, the Promise from wsk.rules.list fails but is not used the function response and so the openwhisk handler assumes the function returned normally. Once you return the promise, the handler uses that result as the invocation result. 

The wsk.rules.list call fails in both cases but in the first example it's not propagated to the handler.

**Comment by jthomas** [comment]
Thanks for raising the issue that we need further examples about how to instantiate the library on the openwhisk platform. 

I'm going to close this without merging due to the reasons highlighted above but I will now make different changes to fix this issue in the README.

**Pull request closed without merging**